crates/typos-cli/src/bin/typos-cli/report.rs

```diff
@@ -280,6 +286,155 @@ impl Report for PrintJson {
     }
 }
 
+#[derive(Debug)]
+pub(crate) struct PrintSarif {
+    results: Mutex<RefCell<Vec<sarif::Result>>>,
+    error: Mutex<RefCell<Vec<String>>>,
```

Comment:
Why do we need `RefCell` when we have a `Mutex`?

Reply:
Since `report` in `Report` has a immutable `self`, without `RefCell` I can write to `results` and `error` in the `report`.

Reply:
I recreated this in the playground:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4f989848d30f923f7c3ccc3b5d3c6ac6

I then removed `RefCell` and it still works:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=fad6216927b55e6b98435cba8209abb9

`Mutex` fills a similar role to `RefCell` except its thread-safe.

Reply:
Well it did work, I know so little about rust

Reply:
ok